core/src/main/java/io/kestra/core/tasks/flows/WaitFor.java

```diff
@@ -0,0 +1,238 @@
+
+                return Instant.now().plus(this.interval);
+            }
+        }
+
+        return null;
+    }
+
+    private Boolean reachedMaximums(RunContext runContext, Execution execution, TaskRun parentTaskRun, Boolean printLog) {
+        TaskRun childTaskRun = this.getChildTaskRun(execution, parentTaskRun).orElse(null);
+        Logger logger = runContext.logger();
+
+        if (childTaskRun == null) {
+            return false;
+        }
+
+        if (this.maxIterations != null && childTaskRun.attemptNumber() >= this.maxIterations) {
+            if (printLog) {logger.warn("Max iterations reached");}
+            return true;
+        }
+
+        if (this.maxDuration != null &&
+            childTaskRun.getAttempts().getFirst().getState().getStartDate().plus(this.maxDuration).isBefore(Instant.now())) {
+            if (printLog) {logger.warn("Max duration reached");}
+
+            return true;
+        }
+
+        return false;
+    }
+
+    @Override
+    public Optional<State.Type> resolveState(RunContext runContext, Execution execution, TaskRun parentTaskRun) throws IllegalVariableEvaluationException {
+        TaskRun childTaskRun = this.getChildTaskRun(execution, parentTaskRun).orElse(null);
+        if (childTaskRun != null && nextExecutionDate(runContext, execution, parentTaskRun) != null) {
+            return Optional.of(State.Type.RUNNING);
+        }
+
+        if (childTaskRun != null && this.reachedMaximums(runContext, execution, parentTaskRun, true) && this.failOnMaxReached) {
+            return Optional.of(State.Type.FAILED);
+        }
+
+        return FlowableUtils.resolveState(
+            execution,
+            this.childTasks(runContext, parentTaskRun),
+            FlowableUtils.resolveTasks(this.getErrors(), parentTaskRun),
+            parentTaskRun,
+            runContext,
+            isAllowFailure()
+        );
+    }
+
+    public Optional<TaskRun> getChildTaskRun(Execution execution, TaskRun parentTaskRun) {
+        if (execution.getTaskRunList() == null) {
+            return Optional.empty();
+        }
+        return execution
+            .getTaskRunList()
+            .stream()
+            .filter(t -> t.getParentTaskRunId() != null && t.getParentTaskRunId().equals(parentTaskRun.getId()) && t.getState().isSuccess())
+            .findFirst();
+    }
+
+    @Override
+    public WaitFor.Output outputs(RunContext runContext, Execution execution, TaskRun parentTaskRun) throws IllegalVariableEvaluationException {
+        String value = parentTaskRun != null ?
+            parentTaskRun.getOutputs().get("iterationCount").toString() : "0";
+
+        return Output.builder()
+            .iterationCount(Integer.parseInt(value) + 1)
+            .build();
+    }
+
+    @Builder
+    @Getter
+    public static class Output implements io.kestra.core.models.tasks.Output {
+        private Integer iterationCount;
+        private Map<String, Object> lastOutputs;
```

Comment:
```suggestion
        private Map<String, Object> finalOutputs;
```

I think `final` sounds better here - WDYT?

Reply:
Its not used anymore so I will just remove it (but final would have make the job!)